pkg/types/validation/installconfig.go

```diff
@@ -196,7 +196,7 @@ func validateNetworkingIPVersion(n *types.Networking, p *types.Platform) field.E
 		}
 
 		switch {
-		case p.Azure != nil:
+		case p.Azure != nil && os.Getenv("OPENSHIFT_INSTALL_EXPERIMENTAL_DUAL_STACK") == "true":
```

Comment:
1. Make the environment variable a bit more friendly by supporting more "true" values.
2. Add a warning that an experimental feature is being used.
```
experimentalDualStackEnabled, _ := strconv.ParseBool(os.Getenv("OPENSHIFT_INSTALL_EXPERIMENTAL_DUAL_STACK"))
switch {
case p.Azure != nil && experimentalDualStackEnabled:
	fmt.Warn("Using experimental dual-stack support")
```